sdk/include/platform/ibex/platform-dma.hh

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#define MALLOC_QUOTA 0x100000
+#define TEST_NAME "Allocator"
+#include "tests.hh"
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <cstdint>
+#include <stdint.h>
+#include <utils.hh>
+
+#include <thread.h>
+#include <thread_pool.h>
+
+using thread_pool::async;
+DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(dmaDriverHeap, 8144);
+
+namespace Ibex {
+    class GenericDMA {
+        private:
+
+        Timeout noWait{0};
+        SObjStruct *dmaHeap;
+
+        struct StreamingInterface
+        {
+            uint32_t control;
+            uint32_t status;
+            uint32_t address;
+            uint32_t lengthInBytes;
+        };
+
+        struct DMAInterface
+        {
+            /**
+             *  these two streaming interfaces are for 
+             *  [0] memory to target and 
+             *  [1] target to memory directions.
+             *  target here is any I/O device, peripheral or an accelerator
+             */
+            volatile StreamingInterface *streamingInterface[2];
+        };
+
+        volatile DMAInterface *dmaInterface;
+
+        public:
+
+        void init()
+        {
+            dmaInterface->streamingInterface[0] = MMIO_CAPABILITY(StreamingInterface, dma_to_target);
+            dmaInterface->streamingInterface[1] = MMIO_CAPABILITY(StreamingInterface, dma_to_memory);
+        }
+
+        uint32_t read_status(bool index)
+        {   
+            /**
+             *  this statement returns the less signifance bit 
+             *  to show the halted status
+             */
+            return dmaInterface->streamingInterface[index]->status & 0x1;
+        }
+
+        void write_conf(bool index, uint32_t *address, uint32_t lengthInBytes)
```

Comment:
`bool` seems a strange type for a thing called `index`.  What does it mean?  Looking at the code, I think this is the device to configure - it would probably be better to expose an interface that configures both sides at the same time, or you have fun race conditions if two things both want to start a DMA at the same time.